armory/scenarios/librispeech_dev_clean_spectrogram.py

```diff
@@ -0,0 +1,171 @@
+                    x_seg.append(xt[:, rand_t : rand_t + n_tbins])
+                x_seg = np.array(x_seg)
+                x_seg = np.expand_dims(x_seg, -1)
+
+                classifier.fit(
+                    x_seg,
+                    y,
+                    batch_size=config["dataset"]["batch_size"],
+                    nb_epochs=1,
+                    verbose=True,
+                )
+
+                if cnt % train_data_generator.batches_per_epoch == 0:
+                    # evaluate on validation examples
+                    val_data_generator = load_dataset(
+                        config["dataset"],
+                        epochs=1,
+                        split_type="validation",
+                        preprocessing_fn=preprocessing_fn,
+                    )
+
+                    cnt = 0
+                    validation_accuracies = []
+                    for x_val, y_val in tqdm(val_data_generator):
+                        x_val_seg, y_val_seg = [], []
+                        for xt, yt in zip(x_val, y_val):
+                            n_seg = int(xt.shape[1] / n_tbins)
+                            xt = xt[:, : n_seg * n_tbins]
+                            for ii in range(n_seg):
+                                x_val_seg.append(
+                                    xt[:, ii * n_tbins : (ii + 1) * n_tbins]
+                                )
+                                y_val_seg.append(yt)
+                        x_val_seg = np.array(x_val_seg)
+                        x_val_seg = np.expand_dims(x_val_seg, -1)
+                        y_val_seg = np.array(y_val_seg)
+
+                        y_pred = classifier.predict(x_val_seg)
+                        validation_accuracies.extend(task_metric(y_val_seg, y_pred))
+                        cnt += len(y_val_seg)
+                    validation_accuracy = sum(validation_accuracies) / cnt
+                    logger.info("Validation accuracy: {}".format(validation_accuracy))
+
+        classifier.set_learning_phase(False)
+        # Evaluate ART classifier on test examples
+        logger.info(f"Loading testing dataset {config['dataset']['name']}...")
+        test_data_generator = load_dataset(
+            config["dataset"],
+            epochs=1,
+            split_type="test",
+            preprocessing_fn=preprocessing_fn,
+        )
+
+        logger.info("Running inference on benign test examples...")
+
+        cnt = 0
+        benign_accuracies = []
+        for x, y in tqdm(test_data_generator, desc="Benign"):
+            x_seg, y_seg = [], []
+            for xt, yt in zip(x, y):
+                n_seg = int(xt.shape[1] / n_tbins)
+                xt = xt[:, : n_seg * n_tbins]
+                for ii in range(n_seg):
+                    x_seg.append(xt[:, ii * n_tbins : (ii + 1) * n_tbins])
+                    y_seg.append(yt)
+            x_seg = np.array(x_seg)
+            x_seg = np.expand_dims(x_seg, -1)
+            y_seg = np.array(y_seg)
```

Comment:
Since this repeatedly shows up, I would recommend cleaning it up and making a function out of it:
```
def segment(x, y, n_time_bins):
    """Return segmented batch with dims ..."""
    ...
    return x_seg, y_seg
```